soda_machine.rb

```diff
@@ -7,12 +7,22 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
```

Comment:
Logic is sound but there's an excessive amount of code here. Try reimplementing this with single line `.find` call.